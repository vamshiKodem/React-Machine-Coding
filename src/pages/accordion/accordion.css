.accordion-container {
    padding: 20px 50px;

}

.accordion-heading {
    height: 30px;
    width: 50%;
    background-color: khaki;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0px 20px;
    cursor: pointer;
}

.accordion-description {
    padding: 10px 20px;
    width: 50%;
    background-color: rgb(199, 190, 112);
}