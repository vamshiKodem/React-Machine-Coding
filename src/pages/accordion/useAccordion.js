import { useState } from "react";

export const useAccordion = () => {
  const [isVisible, setIsVisible] = useState(false);

  const onTitleClick = () => {
    setIsVisible((prev) => !prev);
  };

  return {
    isVisible,
    onTitleClick,
  };
};
