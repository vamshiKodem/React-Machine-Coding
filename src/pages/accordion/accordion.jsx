import React from "react";
import "./accordion.css";
import { useAccordion } from "./useAccordion";

const Accordion = ({ title, description }) => {
  const { isVisible, onTitleClick } = useAccordion();
  return (
    <div className="accordion-container">
      <div className="accordion-heading" onClick={onTitleClick}>
        <h2>{title}</h2>
        <span>{isVisible ? "-" : "+"}</span>
      </div>
      {isVisible ? (
        <div className="accordion-description">{description}</div>
      ) : null}
    </div>
  );
};

export const AccordionUI = () => {
  const accordionData = [
    {
      title: "Section 1",
      content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Quis sapiente
          laborum cupiditate possimus labore, hic temporibus velit dicta earum
          suscipit commodi eum enim atque at? Et perspiciatis dolore iure
          voluptatem.`,
    },
    {
      title: "Section 2",
      content: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia veniam
          reprehenderit nam assumenda voluptatem ut. Ipsum eius dicta, officiis
          quaerat iure quos dolorum accusantium ducimus in illum vero commodi
          pariatur? Impedit autem esse nostrum quasi, fugiat a aut error cumque
          quidem maiores doloremque est numquam praesentium eos voluptatem amet!
          Repudiandae, mollitia id reprehenderit a ab odit!`,
    },
    {
      title: "Section 3",
      content: `Sapiente expedita hic obcaecati, laboriosam similique omnis architecto ducimus magnam accusantium corrupti
          quam sint dolore pariatur perspiciatis, necessitatibus rem vel dignissimos
          dolor ut sequi minus iste? Quas?`,
    },
  ];
  return (
    <>
      {accordionData.map((accordion) => (
        <Accordion
          key={accordion.title}
          title={accordion.title}
          description={accordion.content}          
        />
      ))}
    </>
  );
};
