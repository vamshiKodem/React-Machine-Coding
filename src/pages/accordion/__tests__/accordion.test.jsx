import React from "react";
import { render } from "@testing-library/react";
import { AccordionUI } from "../accordion";

test("renders button with correct text", () => {
  const { container } = render(<AccordionUI />);
  expect(container).toMatchSnapshot();
});
