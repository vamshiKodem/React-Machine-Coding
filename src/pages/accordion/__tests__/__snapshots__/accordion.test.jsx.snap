// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with correct text 1`] = `
<div>
  <div
    class="accordion-container"
  >
    <div
      class="accordion-heading"
    >
      <h2>
        Section 1
      </h2>
      <span>
        +
      </span>
    </div>
  </div>
  <div
    class="accordion-container"
  >
    <div
      class="accordion-heading"
    >
      <h2>
        Section 2
      </h2>
      <span>
        +
      </span>
    </div>
  </div>
  <div
    class="accordion-container"
  >
    <div
      class="accordion-heading"
    >
      <h2>
        Section 3
      </h2>
      <span>
        +
      </span>
    </div>
  </div>
</div>
`;
